app/components/AiChatBase.tsx

```diff
@@ -0,0 +1,156 @@
+'use client';
+
+import Image from 'next/image';
+import { PropsWithChildren, useEffect, useState } from 'react';
+import * as Dialog from '@radix-ui/react-dialog';
+import { cn, MarkdownRenderer, ScrollArea } from '@/app/components';
+import aiStarsSvg from '@/public/ai-stars.svg';
+import wizardSvg from '@/public/pixel-wizard.svg';
+import { Icon, IconButton, Input, Tag } from '@swapr/ui';
+import { twMerge } from 'tailwind-merge';
+import { FA_EVENTS } from '@/analytics';
+import { trackEvent } from 'fathom-client';
+
+interface AiBaseProps extends PropsWithChildren {
+  messages: any[];
+  trackOnClickEvents: string[];
+  isChatOpen?: boolean;
+}
+
+const MARKETING_LINK =
+  'https://swpr.notion.site/Presagio-AI-Predictor-Chatbot-Beta-Launch-f4ccdfa867e949d3badf10705b7c90aa';
+
+export const AiChatBase = ({
+  children,
+  messages,
+  trackOnClickEvents,
+  isChatOpen,
+}: AiBaseProps) => {
+  const [isOpen, setOpen] = useState(isChatOpen);
+  const [scrollAreaElement, setScrollAreaElement] = useState<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (scrollAreaElement) scrollAreaElement.scrollTop = scrollAreaElement.scrollHeight;
+  }, [messages, isOpen, scrollAreaElement]);
+
+  return (
+    <Dialog.Root modal onOpenChange={isOpen => setOpen(isOpen)} open={isOpen}>
+      <div className="fixed bottom-4 flex w-full items-center justify-end px-2 md:bottom-10 md:px-6">
+        <div className="flex flex-col items-end">
+          <Dialog.Portal>
+            <Dialog.Content className="fixed bottom-20 right-0 w-full origin-bottom-right rounded-16 border border-outline-base-em bg-surface-surface-0 shadow-2 data-[state=open]:animate-grow md:bottom-28 md:right-4 md:w-[420px]">
```

Comment:
Check the comments on this file for further context on the main function

```suggestion
            <Dialog.Content className="fixed bottom-20 right-0 w-full origin-bottom-right rounded-16 border shadow-2 data-[state=open]:animate-grow md:bottom-28 md:right-4 md:w-[420px]">
```